tests/test_dataset.py

```diff
@@ -0,0 +1,520 @@
+import filecmp
+
+import numpy as np
+from shutil import rmtree, copytree
+
+from wkcuber.api.Dataset import WKDataset, TiffDataset
+from os import path, mkdir
+
+from wkcuber.api.Properties import Properties, TiffProperties
+
+
+def delete_dir(relative_path):
+    if path.exists(relative_path) and path.isdir(relative_path):
+        rmtree(relative_path)
+
+
+def test_create_WKDataset_with_layer_and_mag():
+    delete_dir("../testoutput/WKDataset")
+
+    ds = WKDataset.create("../testoutput/WKDataset", scale=(1, 1, 1))
+    ds.add_layer("color", "color")
+
+    ds.get_layer("color").add_mag("1")
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists("../testoutput/WKDataset/color/1")
+    assert path.exists("../testoutput/WKDataset/color/2-2-1")
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+
+def test_create_WKDataset_with_explicit_header_fields():
+    delete_dir("../testoutput/WKDataset_advanced")
+
+    ds = WKDataset.create("../testoutput/WKDataset_advanced", scale=(1, 1, 1))
+    ds.add_layer("color", "color", dtype=np.uint16, num_channels=3)
+
+    ds.get_layer("color").add_mag("1", block_len=64, file_len=64)
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists("../testoutput/WKDataset_advanced/color/1")
+    assert path.exists("../testoutput/WKDataset_advanced/color/2-2-1")
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+    assert ds.properties.data_layers["color"].element_class == np.dtype(np.uint16)
+    assert (
+        ds.properties.data_layers["color"].wkw_resolutions[0].cube_length == 64 * 64
+    )  # mag "1"
```

Comment:
Can the `mag=1` value be read from `wkw_resolutions[0]`? Also, `wkw_magnifications[0]` might be a better name? What do you think?

Reply:
Yes, the mag can be read from `wkw_resolutions[0]`. I have added this to the test